[v2.0] tsconfig with Isolated Modules doesn't like the environment.js file

<!-- This template is for bugs relating to Ember.js typescript support & infrastructure.
     Please fill out all of the required information below -->

### Please paste the output of `ember -v` here
ember-cli: 3.5.1
node: 10.14.1
os: darwin x64

### Please paste the output of `tsc -v` here
Version 3.2.2

### Please paste your `tconfig.json` and `tslint.json` or `eslint.json` (if applicable) below


<details><summary><b>My tsconfig.json</b></summary><pre>

  {
  "compilerOptions": {
    "target": "es2017",
    "allowJs": true,
    "moduleResolution": "node",
    "isolatedModules": true,
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "alwaysStrict": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noEmitOnError": false,
    "noEmit": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "baseUrl": ".",
    "module": "es6",
    "paths": {
      "isolated-modules-error/tests/*": [
        "tests/*"
      ],
      "isolated-modules-error/*": [
        "app/*"
      ],
      "*": [
        "types/*"
      ]
    }
  },
  "include": [
    "app/**/*",
    "tests/**/*",
    "types/**/*"
  ]
}

  
</pre></details>



### What are instructions we can follow to reproduce the issue?
```sh
ember new sample; cd ./sample 
ember install ember-cli-typescript@next 

ember s 
```

### What did you expect?
I expected to not see any type errors right off the bat.

(I did expect a build error, because I haven't yet installed babel, but the type error I am reporting occurs regardless of babel understanding TS. This is just the minimum number of steps to reproduce.)

### What happened instead?
TypeScript complained about this:

```sh
config/environment.js:1:1 - error TS1208: Cannot compile namespaces when the '--isolatedModules' flag is provided.

1 'use strict';
  ~~~~~~~~~~~~
```

I tried:

1. removing 'use strict', but then it just complains about `module.exports`.
1. changing that environment file to a TS file with and export, but that failed to run my after it was built.

So the type error is preferable, but just letting you know that the out-of-the-box types are confused about that env file.

--- a/blueprint-files/ember-cli-typescript/tsconfig.json
+++ b/blueprint-files/ember-cli-typescript/tsconfig.json
@@ -3,7 +3,6 @@
     "target": "es2017",
     "allowJs": true,
     "moduleResolution": "node",
-    "isolatedModules": true,
     "allowSyntheticDefaultImports": true,
     "noImplicitAny": true,
     "noImplicitThis": true,

--- a/ts/tests/blueprints/ember-cli-typescript-test.js
+++ b/ts/tests/blueprints/ember-cli-typescript-test.js
@@ -72,7 +72,6 @@ describe('Acceptance: ember-cli-typescript generator', function() {
 
         expect(tsconfigJson.compilerOptions.inlineSourceMap).to.equal(true);
         expect(tsconfigJson.compilerOptions.inlineSources).to.equal(true);
-        expect(tsconfigJson.compilerOptions.isolatedModules).to.equal(true);
 
         expect(tsconfigJson.include).to.deep.equal(['app/**/*', 'tests/**/*', 'types/**/*']);
 

